diff -Nur a/feeds/luci/applications/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock b/feeds/luci/applications/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock
--- a/feeds/luci/applications/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock	2015-09-06 20:15:10.963638010 +0800
+++ b/feeds/luci/applications/luci-app-p2pblock/root/etc/uci-defaults/luci-p2pblock	2015-09-05 06:49:10.000000000 +0800
@@ -7,5 +7,7 @@
 	commit ucitrack
 EOF
 
+/etc/init.d/freifunk-p2pblock disable
+
 rm -f /tmp/luci-indexcache
 exit 0
diff -Nur a/feeds/luci/modules/luci-base/po/zh-cn/base.po b/feeds/luci/modules/luci-base/po/zh-cn/base.po
--- a/feeds/luci/modules/luci-base/po/zh-cn/base.po	2015-09-06 20:15:11.112638010 +0800
+++ b/feeds/luci/modules/luci-base/po/zh-cn/base.po	2015-09-04 12:44:24.000000000 +0800
@@ -2776,7 +2776,7 @@
 msgstr "使用广播标签"
 
 msgid "Use builtin IPv6-management"
-msgstr ""
+msgstr "使用内置IPv6管理"
 
 msgid "Use custom DNS servers"
 msgstr "使用自定义的DNS服务器"
diff -Nur a/feeds/luci/modules/luci-base/root/etc/config/luci b/feeds/luci/modules/luci-base/root/etc/config/luci
--- a/feeds/luci/modules/luci-base/root/etc/config/luci	2015-09-06 20:15:11.112638010 +0800
+++ b/feeds/luci/modules/luci-base/root/etc/config/luci	2015-09-06 20:17:10.093637876 +0800
@@ -1,6 +1,6 @@
 config core main
-	option lang auto
-	option mediaurlbase /luci-static/openwrt.org
+	option lang zh_cn
+	option mediaurlbase /luci-static/bootstrap
 	option resourcebase /luci-static/resources
 	
 config extern flash_keep
@@ -22,3 +22,4 @@
 	option enable 1
 		
 config internal themes
+	option Bootstrap "/luci-static/bootstrap"
diff -Nur a/feeds/packages/net/sqm-scripts/files/usr/lib/lua/luci/i18n/sqm.zh-cn.po b/feeds/packages/net/sqm-scripts/files/usr/lib/lua/luci/i18n/sqm.zh-cn.po
--- a/feeds/packages/net/sqm-scripts/files/usr/lib/lua/luci/i18n/sqm.zh-cn.po	1970-01-01 08:00:00.000000000 +0800
+++ b/feeds/packages/net/sqm-scripts/files/usr/lib/lua/luci/i18n/sqm.zh-cn.po	2015-09-05 12:13:12.000000000 +0800
@@ -0,0 +1,139 @@
+msgid ""
+msgstr ""
+"Project-Id-Version: \n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2015-09-04\n"
+"PO-Revision-Date: 2015-09-05 12:13+0800\n"
+"Last-Translator: kuoruan <kuoruan@qq.com>\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=1; plural=0;\n"
+"X-Generator: Poedit 1.8.4\n"
+"Language: zh_CN\n"
+"Language-Team: \n"
+
+msgid "Smart Queue Management"
+msgstr "智能队列管理"
+
+msgid ""
+"With <abbr title=\"Smart Queue Management\">SQM</abbr> you can enable traffic shaping, better "
+"mixing (Fair Queueing), active queue length management (AQM) and prioritisation on one network "
+"interface."
+msgstr ""
+"使用 <abbr title=\"智能队列管理\">SQM</abbr>，你可以启用流量控制、 更好地混合 (队列)、 主动管理队列"
+"长度(AQM) 和一个网络接口的优先级。"
+
+msgid "Queues"
+msgstr "队列"
+
+msgid "Basic Settings"
+msgstr "基本设置"
+
+msgid "Queue Discipline"
+msgstr "队列规则"
+
+msgid "Link Layer Adaptation"
+msgstr "链路层适配"
+
+msgid "Enable this SQM instance."
+msgstr "启用 SQM"
+
+msgid ""
+"The SQM GUI has just enabled the sqm initscript on your behalf. Remember to disable the sqm "
+"initscript manually under System Startup menu in case this change was not wished for."
+msgstr ""
+"SQM GUI 刚刚在你的操作下启用了 SQM 初始化脚本，如果不希望应用此更改，请在系统启动菜单中手动禁用此 "
+"SQM 初始化脚本。"
+
+msgid "Interface name"
+msgstr "接口名称"
+
+msgid "Download speed (kbit/s) (ingress) set to 0 to selectively disable ingress shaping:"
+msgstr "下载速度(kbit/s), 0 不限制"
+
+msgid "Upload speed (kbit/s) (egress) set to 0 to selectively disable egress shaping:"
+msgstr "上传速度(kbit/s), 0 不限制"
+
+msgid ""
+"Queuing disciplines useable on this system; instantiated only after first successful start of SQM. "
+"You need to start a new GUI session to see updates!"
+msgstr "使用队列规则"
+
+msgid "default"
+msgstr "默认"
+
+msgid "Queue setup script"
+msgstr "设置队列脚本"
+
+msgid ""
+"Show and Use Advanced Configuration. Advanced options will only be used as long as this box is "
+"checked."
+msgstr "显示和使用高级配置项。选中该复选框，将只会使用高级选项"
+
+msgid "Squash DSCP on inbound packets (ingress):"
+msgstr "入站数据包 DSCP 优先级"
+
+msgid "Ignore DSCP on ingress:"
+msgstr "忽略的 DSCP 入站数据包"
+
+msgid "Explicit congestion notification (ECN) status on inbound packets (ingress):"
+msgstr "在入站数据包上显示拥塞通知 (ECN) 状态"
+
+msgid "Explicit congestion notification (ECN) status on outbound packets (egress)."
+msgstr "在出站数据包上显示拥塞通知 (ECN) 状态"
+
+msgid ""
+"Show and Use Dangerous Configuration. Dangerous options will only be used as long as this box is "
+"checked."
+msgstr "显示和使用危险配置项。选中该复选框，将只会使用危险选项"
+
+msgid "Hard limit on ingress queues; leave empty for default."
+msgstr "强制进入入站队列，留空以使用默认"
+
+msgid "Hard limit on egress queues; leave empty for default."
+msgstr "强制进入出站队列，留空以使用默认"
+
+msgid ""
+"Latency target for ingress, e.g 5ms [units: s, ms, or us]; leave empty for automatic selection, "
+"put in the word default for the qdisc's default."
+msgstr "入站目标延迟，例如5ms[单位：s,ms,us]，留空以自动选择"
+
+msgid ""
+"Latency target for egress, e.g. 5ms [units: s, ms, or us]; leave empty for automatic selection, "
+"put in the word default for the qdisc's default."
+msgstr "出站目标延迟，例如5ms[单位：s,ms,us]，留空以自动选择"
+
+msgid ""
+"Advanced option string to pass to the ingress queueing disciplines; no error checking, use very "
+"carefully."
+msgstr "将高级选项字符串传递到入站队列规则，无错误检查，谨慎使用"
+
+msgid ""
+"Advanced option string to pass to the egress queueing disciplines; no error checking, use very "
+"carefully."
+msgstr "将高级选项字符串传递到出站队列规则，无错误检查，谨慎使用"
+
+msgid "Which link layer to account for:"
+msgstr "链路层适配"
+
+msgid "Per Packet Overhead (byte):"
+msgstr "每数据包开销 (byte)"
+
+msgid ""
+"Show Advanced Linklayer Options, (only needed if MTU > 1500). Advanced options will only be used "
+"as long as this box is checked."
+msgstr "显示高级链路层选项，(只在 MTU > 1500 时需要)。当这个选项被选中，将只使用高级选项"
+
+msgid ""
+"Maximal Size for size and rate calculations, tcMTU (byte); needs to be >= interface MTU + overhead:"
+msgstr "使用最大的大小和速度计算 tcMTU (byte)；需要 >= 接口的 MTU + 开销"
+
+msgid "Number of entries in size/rate tables, TSIZE; for ATM choose TSIZE = (tcMTU + 1) / 16:"
+msgstr "大小/费率表中的条目数 TSIZE；对于 ATM 选择 TSIZE = (tcMTU + 1) / 16"
+
+msgid "Minimal packet size, MPU (byte); needs to be > 0 for ethernet size tables:"
+msgstr "最小的数据包大小 MPU (byte)，需要以太网尺寸表 > 0"
+
+msgid "Which linklayer adaptation mechanism to use; for testing only"
+msgstr "使用链路自适应机制，仅用于测试"
diff -Nur a/feeds/packages/net/sqm-scripts/Makefile b/feeds/packages/net/sqm-scripts/Makefile
--- a/feeds/packages/net/sqm-scripts/Makefile	2015-09-06 20:14:23.539638064 +0800
+++ b/feeds/packages/net/sqm-scripts/Makefile	2015-09-04 20:52:06.000000000 +0800
@@ -19,6 +19,8 @@
 PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)-$(PKG_RELEASE)
 PKG_BUILD_DIR := $(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)-$(PKG_RELEASE)
 
+PO2LMO:=$(STAGING_DIR_HOST)/bin/po2lmo
+
 include $(INCLUDE_DIR)/package.mk
 
 define Package/sqm-scripts
@@ -42,6 +44,9 @@
 
 define Package/sqm-scripts/install
 	make -C $(PKG_BUILD_DIR) DESTDIR=$(1) PLATFORM=openwrt install
+	$(CP) ./files/* $(1)
+	${PO2LMO} $(1)/usr/lib/lua/luci/i18n/sqm.zh-cn.po $(1)/usr/lib/lua/luci/i18n/sqm.zh-cn.lmo
+	rm $(1)/usr/lib/lua/luci/i18n/sqm.zh-cn.po
 endef
 
 define Package/luci-app-sqm
diff -Nur a/package/kernel/mac80211/files/lib/wifi/mac80211.sh b/package/kernel/mac80211/files/lib/wifi/mac80211.sh
--- a/package/kernel/mac80211/files/lib/wifi/mac80211.sh	2015-09-06 18:50:12.780070565 +0800
+++ b/package/kernel/mac80211/files/lib/wifi/mac80211.sh	2015-09-06 20:17:10.093637876 +0800
@@ -128,7 +128,7 @@
 	option device   radio$devidx
 	option network  lan
 	option mode     ap
-	option ssid     OpenWrt${ssid_5ghz}-$(cat /sys/class/ieee80211/${dev}/macaddress | awk -F ":" '{print $4""$5""$6 }'| tr a-z A-Z)
+	option ssid     HiWiFi${ssid_5ghz}-$(cat /sys/class/ieee80211/${dev}/macaddress | awk -F ":" '{print $4""$5""$6 }'| tr a-z A-Z)
 	option encryption none
 
 EOF
diff -Nur a/target/linux/ramips/base-files/etc/board.d/01_leds b/target/linux/ramips/base-files/etc/board.d/01_leds
--- a/target/linux/ramips/base-files/etc/board.d/01_leds	2015-09-06 18:50:13.161070565 +0800
+++ b/target/linux/ramips/base-files/etc/board.d/01_leds	2015-09-06 20:17:10.100637873 +0800
@@ -224,6 +224,11 @@
 wt3020)
 	ucidef_set_led_default "power" "power" "$board:blue:power" "0"
 	;;
+HC5661)
+	ucidef_set_led_default "power" "power" "hiwifi:blue:power" "1"
+	ucidef_set_led_netdev "internet" "internet" "hiwifi:blue:internet" "eth0.2"
+	set_wifi_led "hiwifi:blue:wifi"
+	;;
 y1|\
 y1s)
 	ucidef_set_led_default "power" "power" "$board:blue:power" "1"
diff -Nur a/target/linux/ramips/base-files/etc/board.d/02_network b/target/linux/ramips/base-files/etc/board.d/02_network
--- a/target/linux/ramips/base-files/etc/board.d/02_network	2015-09-06 18:50:13.161070565 +0800
+++ b/target/linux/ramips/base-files/etc/board.d/02_network	2015-09-06 20:17:10.100637873 +0800
@@ -217,6 +217,7 @@
 		ucidef_add_switch_vlan "switch0" "1" "0 1 2 3 6t"
 		ucidef_add_switch_vlan "switch0" "2" "5 6t"
 		;;
+	HC5661 |\
 	y1s)
 		ucidef_set_interfaces_lan_wan "eth0.1" "eth0.2"
 		ucidef_add_switch "switch0" "1" "1"
@@ -267,6 +268,12 @@
 		lan_mac=$(macaddr_setbit_la "$lan_mac")
 		wan_mac=$(macaddr_add "$lan_mac" 1)
 		;;
+	HC5661)
+		__fac_mac=`strings /dev/mtd7 | grep 'fac_mac = ..:..:..:..:..:..'`
+		lan_mac=`expr "$__fac_mac" : '.*\(..:..:..:..:..:..\)' | tr '[A-Z]' '[a-z]'`
+		[ -n "$lan_mac" ] || lan_mac=$(cat /sys/class/net/eth0/address)
+		wan_mac=$(macaddr_add "$lan_mac" 1)
+		;;
 	all0239-3g|\
 	carambola|\
 	freestation5|\
diff -Nur a/target/linux/ramips/base-files/etc/diag.sh b/target/linux/ramips/base-files/etc/diag.sh
--- a/target/linux/ramips/base-files/etc/diag.sh	2015-09-06 18:50:13.161070565 +0800
+++ b/target/linux/ramips/base-files/etc/diag.sh	2015-09-06 20:17:10.101637875 +0800
@@ -90,6 +90,9 @@
 	wl-330n3g|\
 	wli-tx4-ag300n|\
 	wt3020|\
+	HC5661)
+		status_led="hiwifi:blue:power"
+		;;
 	y1|\
 	y1s)
 		status_led="$board:blue:power"
diff -Nur a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
--- a/target/linux/ramips/base-files/lib/ramips.sh	2015-09-06 18:50:13.162070565 +0800
+++ b/target/linux/ramips/base-files/lib/ramips.sh	2015-09-06 20:17:10.101637875 +0800
@@ -418,6 +418,9 @@
 	*"X8")
 		name="x8"
 		;;
+	*"HiWiFi HC5661")
+		name="HC5661"
+		;;
 	*"Y1")
 		name="y1"
 		;;
diff -Nur a/target/linux/ramips/base-files/lib/upgrade/platform.sh b/target/linux/ramips/base-files/lib/upgrade/platform.sh
--- a/target/linux/ramips/base-files/lib/upgrade/platform.sh	2015-09-06 18:50:13.162070565 +0800
+++ b/target/linux/ramips/base-files/lib/upgrade/platform.sh	2015-09-06 20:17:10.101637875 +0800
@@ -123,6 +123,7 @@
 	wzr-agl300nh|\
 	x5|\
 	x8|\
+	HC5661|\
 	y1|\
 	y1s|\
 	zbt-wa05|\
diff -Nur a/target/linux/ramips/dts/HC5661.dts b/target/linux/ramips/dts/HC5661.dts
--- a/target/linux/ramips/dts/HC5661.dts	1970-01-01 08:00:00.000000000 +0800
+++ b/target/linux/ramips/dts/HC5661.dts	2015-09-06 20:17:10.102637876 +0800
@@ -0,0 +1,157 @@
+/dts-v1/;
+
+/include/ "mt7620a.dtsi"
+
+/ {
+	compatible = "HiWiFi-HC5661", "ralink,mt7620a-soc";
+	model = "HiWiFi HC5661";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+	
+	palmbus@10000000 {
+		sysc@0 {
+			ralink,gpiomux = "i2c", "jtag";
+			ralink,uartmux = "gpio";
+			ralink,wdtmux = <1>;
+		};
+		gpio0: gpio@600 {
+			status = "okay";
+		};
+		gpio2: gpio@660 {
+			status = "okay";
+		};
+		gpio3: gpio@688 {
+			status = "okay";
+		};
+
+		spi@b00 {
+			status = "okay";
+
+			m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "w25q128";
+				reg = <0 0>;
+				linux,modalias = "m25p80", "w25q128";
+				spi-max-frequency = <10000000>;
+
+				partition@0 {
+					label = "u-boot";
+					reg = <0x0 0x30000>;
+				};
+
+				partition@30000 {
+					label = "u-boot-env";
+					reg = <0x30000 0x10000>;
+					read-only;
+				};
+
+				factory: partition@40000 {
+					label = "factory";
+					reg = <0x40000 0x10000>;
+				};
+
+				partition@50000 {
+					label = "firmware";
+					reg = <0x50000 0xf80000>;
+				};
+
+				partition@fd0000 {
+					label = "hwf_config";
+					reg = <0xfd0000 0x10000>;
+				};
+
+				bdinfo: partition@fe0000 {
+					label = "bdinfo";
+					reg = <0xfe0000 0x10000>;
+				};
+
+				partition@ff0000 {
+					label = "backup";
+					reg = <0xff0000 0x10000>;
+				};
+			};
+		};
+	};
+
+	ehci@101c0000 {
+		status = "okay";
+	};
+
+	ohci@101c1000 {
+		status = "okay";
+	};
+
+	ethernet@10100000 {
+		pinctrl-names = "default";
+		pinctrl-0 = <&ephy_pins>;
+		mtd-mac-address = <&factory 0x4>;
+		ralink,port-map = "wllll";
+	};
+
+	sdhci@10130000 {
+		status = "okay";
+	};
+
+	wmac@10180000 {
+		ralink,mtd-eeprom = <&factory 0>;
+	};
+
+	pcie@10140000 {
+		status = "okay";
+	};
+
+	pinctrl {
+		state_default: pinctrl0 {
+			gpio {
+				ralink,group = "uartf", "wled", "nd_sd";
+				ralink,function = "gpio";
+			};
+			pa {
+				ralink,group = "pa";
+				ralink,function = "pa";
+			};
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+		power {
+			label = "hiwifi:blue:power";
+			gpios = <&gpio0 9 1>;
+		};
+		wlan {
+			label = "hiwifi:blue:wifi";
+			gpios = <&gpio3 0 1>;
+		};
+		internet {
+			label = "hiwifi:blue:internet";
+			gpios = <&gpio0 11 1>;
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+		reset {
+			label = "reset";
+			gpios = <&gpio0 12 1>;
+			linux,code = <0x198>;
+		};
+	};
+
+	gpio_export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+		usbpower {
+			gpio-export,name = "usbpower";
+			gpio-export,output = <1>;
+			gpios = <&gpio0 13 0>;
+		};
+	};
+};
+
diff -Nur a/target/linux/ramips/image/Makefile b/target/linux/ramips/image/Makefile
--- a/target/linux/ramips/image/Makefile	2015-09-06 18:50:13.171070565 +0800
+++ b/target/linux/ramips/image/Makefile	2015-09-06 20:17:10.102637876 +0800
@@ -850,74 +850,11 @@
 # MT7620A Profiles
 #
 
-Image/Build/Profile/E1700=$(call BuildFirmware/UMedia/$(1),$(1),e1700,E1700,0x013326)
-Image/Build/Profile/MT7620a=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a,MT7620a)
-Image/Build/Profile/MT7620a_MT7610e=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_mt7610e,MT7620a_MT7610e)
-Image/Build/Profile/MT7620a_MT7530=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_mt7530,MT7620a_MT7530)
-Image/Build/Profile/MT7620a_V22SG=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_v22sg,MT7620a_V22SG)
-br100_mtd_size=8126464
-Image/Build/Profile/AI-BR100=$(call BuildFirmware/CustomFlash/$(1),$(1),ai-br100,AI-BR100,$(br100_mtd_size),Ai-BR)
-Image/Build/Profile/RP-N53=$(call BuildFirmware/Default8M/$(1),$(1),rp-n53,RP-N53)
-whr_300hp2_mtd_size=7012352
-Image/Build/Profile/WHR300HP2=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-300hp2,WHR-300HP2,$(whr_300hp2_mtd_size))
-Image/Build/Profile/WHR600D=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-600d,WHR-600D,$(whr_300hp2_mtd_size))
-whr_1166d_mtd_size=15400960
-Image/Build/Profile/WHR1166D=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-1166d,WHR-1166D,$(whr_1166d_mtd_size))
-dlink810l_mtd_size=6881280
-Image/Build/Profile/CF-WR800N=$(call BuildFirmware/Default8M/$(1),$(1),cf-wr800n,CF-WR800N)
-Image/Build/Profile/DIR-810L=$(call BuildFirmware/CustomFlash/$(1),$(1),dir-810l,DIR-810L,$(dlink810l_mtd_size))
-na930_mtd_size=20971520
-Image/Build/Profile/NA930=$(call BuildFirmware/CustomFlash/$(1),$(1),na930,NA930,$(na930_mtd_size))
-Image/Build/Profile/MZK-750DHP=$(call BuildFirmware/Default8M/$(1),$(1),mzk-750dhp,MZK-750DHP)
-Image/Build/Profile/OY-0001=$(call BuildFirmware/Default16M/$(1),$(1),oy-0001,OY-0001)
-Image/Build/Profile/Y1=$(call BuildFirmware/Default16M/$(1),$(1),y1,Y1)
-Image/Build/Profile/Y1S=$(call BuildFirmware/Default16M/$(1),$(1),y1s,Y1S)
-Image/Build/Profile/MLW221=$(call BuildFirmware/Default16M/$(1),$(1),mlw221,MLW221)
-Image/Build/Profile/MLWG2=$(call BuildFirmware/Default16M/$(1),$(1),mlwg2,MLWG2)
-Image/Build/Profile/WMR-300=$(call BuildFirmware/Default8M/$(1),$(1),wmr-300,WMR-300)
-Image/Build/Profile/RT-N14U=$(call BuildFirmware/Default8M/$(1),$(1),rt-n14u,RT-N14U)
-Image/Build/Profile/WRTNODE=$(call BuildFirmware/Default16M/$(1),$(1),wrtnode,WRTNODE)
-Image/Build/Profile/WT3020=$(call BuildFirmware/PorayDualSize/$(1),$(1),wt3020,WT3020)
-Image/Build/Profile/MIWIFI-MINI=$(call BuildFirmware/Default16M/$(1),$(1),miwifi-mini,MIWIFI-MINI)
-Image/Build/Profile/ZTE-Q7=$(call BuildFirmware/Default8M/$(1),$(1),zte-q7,ZTE-Q7)
-Image/Build/Profile/ZBT-WA05=$(call BuildFirmware/Default8M/$(1),$(1),zbt-wa05,ZBT-WA05)
-Image/Build/Profile/ZBT-WR8305RT=$(call BuildFirmware/Default8M/$(1),$(1),zbt-wr8305rt,ZBT-WR8305RT)
-Image/Build/Profile/ArcherC20i=$(call BuildFirmware/Tplink/$(1),$(1),ArcherC20i,ArcherC20i)
-microwrt_mtd_size=16515072
-Image/Build/Profile/MicroWRT=$(call BuildFirmware/CustomFlash/$(1),$(1),microwrt,MicroWRT,$(microwrt_mtd_size))
-
+Image/Build/Profile/HC5661=$(call BuildFirmware/Default16M/$(1),$(1),hiwifi-hc5661,HC5661)
 
 ifeq ($(SUBTARGET),mt7620)
 define Image/Build/Profile/Default
-	$(call Image/Build/Profile/E1700,$(1))
-	$(call Image/Build/Profile/MT7620a,$(1))
-	$(call Image/Build/Profile/MT7620a_MT7610e,$(1))
-	$(call Image/Build/Profile/MT7620a_MT7530,$(1))
-	$(call Image/Build/Profile/MT7620a_V22SG,$(1))
-	$(call Image/Build/Profile/AI-BR100,$(1))
-	$(call Image/Build/Profile/CF-WR800N,$(1))
-	$(call Image/Build/Profile/RP-N53,$(1))
-	$(call Image/Build/Profile/DIR-810L,$(1))
-	$(call Image/Build/Profile/WHR300HP2,$(1))
-	$(call Image/Build/Profile/WHR600D,$(1))
-	$(call Image/Build/Profile/WHR1166D,$(1))
-	$(call Image/Build/Profile/MZK-750DHP,$(1))
-	$(call Image/Build/Profile/NA930,$(1))
-	$(call Image/Build/Profile/OY-0001,$(1))
-	$(call Image/Build/Profile/Y1,$(1))
-	$(call Image/Build/Profile/Y1S,$(1))
-	$(call Image/Build/Profile/MLW221,$(1))
-	$(call Image/Build/Profile/MLWG2,$(1))
-	$(call Image/Build/Profile/WMR-300,$(1))
-	$(call Image/Build/Profile/RT-N14U,$(1))
-	$(call Image/Build/Profile/WRTNODE,$(1))
-	$(call Image/Build/Profile/WT3020,$(1))
-	$(call Image/Build/Profile/MIWIFI-MINI,$(1))
-	$(call Image/Build/Profile/ZTE-Q7,$(1))
-	$(call Image/Build/Profile/ZBT-WA05,$(1))
-	$(call Image/Build/Profile/ZBT-WR8305RT,$(1))
-	$(call Image/Build/Profile/ArcherC20i,$(1))
-	$(call Image/Build/Profile/MicroWRT,$(1))
+	$(call Image/Build/Profile/HC5661,$(1))
 endef
 endif
 
diff -Nur a/target/linux/ramips/mt7620/profiles/HiWiFi-HC5661.mk b/target/linux/ramips/mt7620/profiles/HiWiFi-HC5661.mk
--- a/target/linux/ramips/mt7620/profiles/HiWiFi-HC5661.mk	1970-01-01 08:00:00.000000000 +0800
+++ b/target/linux/ramips/mt7620/profiles/HiWiFi-HC5661.mk	2015-09-06 20:17:10.102637876 +0800
@@ -0,0 +1,21 @@
+#
+# Copyright (C) 2011 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define Profile/HC5661
+	NAME:=HiWiFi HC5661
+	PACKAGES:=\
+		kmod-usb-core kmod-usb-dwc2 kmod-usb-uhci kmod-usb2 kmod-usb-storage  \
+		kmod-mmc-spi kmod-sdhci kmod-sdhci-mt7620 \
+		block-mount mountd kmod-fs-ext4 ppp-mod-pppoe e2fsprogs \
+		kmod-ledtrig-usbdev fdisk
+endef
+
+define Profile/HC5661/Description
+	Default on SD cards and USB storage support.
+endef
+$(eval $(call Profile,HC5661))
+
